### DISCUSSION TITLE
Show Python code in app mode

### DISCUSSION THREAD

**Discussion by superlopuh** [marimo-team/marimo#792] [Q&A] [answered]
We would like to use Marimo as interactive documentation for our project, and I would like to show the Python code used in the cell when in app mode. The current way to do this is to copy/paste the code into a `mo.md`, but there's a risk of the code in the cell and in the md going out of sync. Is there a button that I'm missing that would let us prevent a cell source from hiding in app mode?

**Comment by akshayka** [comment, marked as answer]
Hi, great question. You can use `--include-code` at the CLI:

`marimo run my_notebook.py --include-code`

Then, when the notebook opens, click the "show code" button in the top right to reveal the code. If it helps, we can consider making code shown by default when `--include-code` is passed.

Example:

![image](https://github.com/marimo-team/marimo/assets/1994308/f68c8ebd-003d-442b-9e92-a8986f53a738)

After clicking the button:

![image](https://github.com/marimo-team/marimo/assets/1994308/449845c6-9e98-4f30-af98-c82e3788a83d)

**Reply by superlopuh** [reply]
Thank you! After asking the question, I discovered the "hide code" button in the edit view, I wonder if the same flag could be used to hide/show code in app mode also?

**Reply by akshayka** [reply]
If you hide code in edit mode, then run in run mode with --include-code, your settings for which cells are shown/hidden should be respected.

Is that good enough? Or are you saying that once you're in run mode, you'd like to further adjust the shown/hidden cells?

**Reply by superlopuh** [reply]
Ah no that's perfect, thank you

**Comment by pkdash** [comment]
Is there way to show code when mounting the marimo app as part of another FastAPI app?

**Reply by mscolnick** [reply]
Yep! `create_asgi_app` includes an option for `include_code=True`
